src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,52 @@
+package command;
+
+import storage.Storage;
+import task.Task;
+import task.TaskList;
+import ui.Ui;
+
+public class DeleteCommand extends Command {
+    private int taskIndex;
+
+    /**
+     * Constructor for Delete Command.
+     *
+     * @param taskIndex index of the task to be deleted.
+     * */
+    public DeleteCommand(int taskIndex) {
+        this.taskIndex = taskIndex;
+    }
+
+    /**
+     * Deletes the task from the current TaskList.
+     *
+     * @param tasks Current TaskList.
+     * @param storage Current Storage.
+     * @param ui Current Ui.
+     * */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```

Comment:
Maybe a more descriptive method name to enhance clarity and readability?
`executeDeleteCommand` for example?